# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/mine/devdoc.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

Index: mine/selector.c
===================================================================
--- mine/selector.c	(revision 58)
+++ mine/selector.c	(working copy)
@@ -111,6 +111,10 @@
 				fullname = "Base Packages";
 			if ( !blanks && !strcmp(fullname, "extra") )
 				fullname = "Additional Packages";
+			if ( !blanks && !strcmp(fullname, "dev") )
+				fullname = "Development Packages";
+			if ( !blanks && !strcmp(fullname, "doc") )
+				fullname = "Documentation Packages";
 
                         snprintf(tmpstring1, 512,
 			        "%*s(%c) %s/", blanks, "",
Index: mine/readdb.c
===================================================================
--- mine/readdb.c	(revision 58)
+++ mine/readdb.c	(working copy)
@@ -145,6 +145,20 @@
 	e->next = d->list; d->list = e; e->content.pkg = p;
 	directory_entry_count++;
 
+	if ( strlen(packagename) > 4 && !strcmp(packagename+strlen(packagename)-4, ":dev") ) {
+		d = get_create_directory(rootdir, "dev");
+		e = calloc(1, sizeof(struct directory_entry));
+		e->next = d->list; d->list = e; e->content.pkg = p;
+		directory_entry_count++;
+	}
+
+	if ( strlen(packagename) > 4 && !strcmp(packagename+strlen(packagename)-4, ":doc") ) {
+		d = get_create_directory(rootdir, "doc");
+		e = calloc(1, sizeof(struct directory_entry));
+		e->next = d->list; d->list = e; e->content.pkg = p;
+		directory_entry_count++;
+	}
+
 	/* read basic info like V and C tags */ 
 	while ( fgets(line, 1024, f) != NULL && strcmp(line,"\027\n") ) {
 		if ( (line_length = strlen(line)) < 3 ) continue;
@@ -283,9 +297,11 @@
 }
 
 char *root_order[] = {
-	"all"
+	"all",
 	"base",
 	"extra",
+	"dev",
+	"doc",
 	NULL
 };
 
@@ -298,7 +314,7 @@
 	}
 
 	if ( v1 == v2 ) return strcmp(n1, n2);
-	return v1 == v2 ? 0 : v1 > v2 ? -1 : 1;
+	return v1 == v2 ? 0 : v1 > v2 ? +1 : -1;
 }
 
 void sort_directory(struct directory *d)
